### ISSUE TITLE
Credo.Check.Warning.UnusedEnumOperation is warning at end of catch blocks

### ISSUE THREAD

**Issue by danferreira** [rrrene/credo#953]
### Precheck

* Proposals for new features should be submitted via: https://github.com/rrrene/credo-proposals
* For bugs, please do a quick search and make sure the bug has not yet been reported here

### Environment

* Credo version (`mix credo -v`): 1.6.3
* Erlang/Elixir version (`elixir -v`): 1.13.2
* Operating system: Linux (Ubuntu)

### What were you trying to do?

Just run `mix credo` with the following example code:

```elixir
defmodule Example do
  def run do
    throw [1, 2, 3, 4]
  catch
    values ->
      Enum.map(values, &(&1 + 1))
  end
end
```

### Expected outcome

No warnings. The same behavior as `rescue` and [other blocks ](https://github.com/rrrene/credo/blob/master/lib/credo/code/block.ex#L10-L17)

Currently, to remove the warning is necessary declare some var and return it:

```elixir 
  catch
    values ->
      hold_my_values = Enum.map(values, &(&1 + 1))
      hold_my_values
  end
```

### Actual outcome

```shell
➜ mix credo lib/example.ex        
Checking 1 source file ...

  Warnings - please take a look                                                                                                                                                   
┃ 
┃ [W] ↗ There should be no unused return values for Enum functions.
┃       lib/example.ex:7:7 #(Example.run)

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.09 seconds (0.05s to load, 0.04s running 69 checks on 1 file)
2 mods/funs, found 1 warning.

Use `mix credo explain` to explain issues, `mix credo --help` for options.
```

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by danferreira** [comment]
> Thanks for reporting this grinning It is now fixed on `master`.
> 
> You can try this by setting the Credo dep to
> 
> ```elixir
> {:credo, github: "rrrene/credo"}
> ```
> 
> Please report back if your issue is solved! +1

Awesome. It works like a charm. 
Thank you
